api/tests/test_user_program_record.rs

```diff
@@ -76,3 +79,67 @@ async fn test_load_pbs_x() {
         None
     )
 }
+
+/// Test the `delete_dangling_records` trigger, which deletes all
+/// user_program_records that are no longer referenced by any other rows and are not pbs
+#[test]
+fn test_delete_dangling() {
+    let runner = QueryRunner::new();
+
+    runner.run_with_conn(|conn| {
+        let user = UserFactory::default().username("user1").insert(&conn);
+        let hw_spec = HardwareSpecFactory::default().name("HW1").insert(&conn);
+        let prog_spec = ProgramSpecFactory::default()
+            .hardware_spec(&hw_spec)
+            .name("Prog1")
+            .insert(&conn);
+
```

Comment:
Can you add two more things here:
1. a 2nd record with at least one stat that matches the PB (e.g. cpu_cycles=1), just to make sure we save _all_ the records that have the min, not just one?
2. A user_program that references one record that has all shit stats, so we know it doesn't try to delete non-PBs that are referenced by `user_programs`